File: test/language/templates/java/templates/SubtypeTemplateArgumentTest.java
package templates;

import static org.junit.Assert.*;

import org.junit.Test;

import templates.subtype_template_argument.Field_uint32;
import templates.subtype_template_argument.Compound;
import templates.subtype_template_argument.Field_Compound;
import templates.subtype_template_argument.SubtypeTemplateArgument;

public class SubtypeTemplateArgumentTest
{
    @Test
    public void bitSizeOf()
    {
        final Field_uint32 field_uint32 = new Field_uint32(10);
        final Field_Compound field_compound = new Field_Compound(new Compound(10));
        final SubtypeTemplateArgument subtypeTemplateArgument = new SubtypeTemplateArgument(field_uint32,
                field_uint32, field_uint32, field_compound, field_compound, field_compound);
        assertEquals(192, subtypeTemplateArgument.bitSizeOf());
    }
}

File: test/language/templates/cpp98/SubtypeTemplateArgumentTest.cpp
#include "gtest/gtest.h"

#include "templates/subtype_template_argument/SubtypeTemplateArgument.h"

namespace templates
{
namespace subtype_template_argument
{

TEST(SubtypeTemplateArgumentTest, bitSizeOf)
{
    Field_uint32 field_uint32;
    field_uint32.setValue(10);

    Compound compound;
    Field_Compound field_compound;
    field_compound.setValue(compound);

    SubtypeTemplateArgument subtypeTemplateArgument;
    subtypeTemplateArgument.setAnotherUint32TypeField(field_uint32);
    subtypeTemplateArgument.setUint32TypeField(field_uint32);
    subtypeTemplateArgument.setUint32Field(field_uint32);
    subtypeTemplateArgument.setAnotherCompoundTypeField(field_compound);
    subtypeTemplateArgument.setCompoundTypeField(field_compound);
    subtypeTemplateArgument.setCompoundField(field_compound);
    ASSERT_EQ(192, subtypeTemplateArgument.bitSizeOf());
}

} // namespace subtype_template_argument
} // namespace templates
